examples/c-hello-world/flake.nix

```diff
@@ -0,0 +1,26 @@
+{
+  description = "A basic flake with a shell";
+
+  inputs.nixpkgs.url = "nixpkgs/nixos-unstable";
+  inputs.flake-utils.url = "github:numtide/flake-utils";
+  # inputs.nickel-nix.url = "github:nickel-lang/nickel-nix";
+  inputs.nickel-nix.url = "path:../../";
```

Comment:
Using "git+file:../../" here seems to fix a `error: cannot fetch input 'path:../../?lastModified=0&narHash=sha256-12fmSIyeZCB5XWAFSIm4AYYVlvI5eQLeg2FXPcbaMsc=' because it uses a relative path`

See https://github.com/NixOS/nix/issues/3978#issuecomment-907573179

Still cannot use things like `nix flake check` and `nix flake show` but seems to get us a little closer.

Reply:
Actually, I get the same error when trying to use `nix build .#` but it goes away when using `git+file` so this appears to have more benefit.

Reply:
Oh, good catch. In fact we set the relative path for development only: in this PR, I think we can just get rid of it and restore the commented input.

Reply:
I think I'll have to do this in two steps. I first fixed the input, but for it to work I need this repo to have the latest version of Nixel, i.e. to merge this PR first. I'll update the inputs of the example in a second step.